### PR TITLE
Protect p2p from vpn tunnel on mobile

### PR DESCRIPTION
**Pull request by anjmao**
Fixes https://github.com/mysteriumnetwork/node/issues/1985

Also:
1. Fixes data race for broker proposals unit tests. (I hope)
2. Some small refactorings, removed textproto conn wrapper.

### CODE REVIEW

File: core/connection/connect_options.go [hunk 1]
```diff
@@ -42,4 +42,5 @@ type ConnectOptions struct {
 	DNS             DNSOption
 	SessionConfig   []byte
 	ProviderNATConn *net.UDPConn
+	P2PFd           int
```
Comment (zolia) [inline core/connection/connect_options.go:45]:
```suggestion
	P2PFD           int
```

File: core/connection/manager_test.go [hunk 2]
```diff
@@ -568,6 +568,10 @@ type mockP2PChannel struct {
 	lock   sync.Mutex
 }
 
+func (m *mockP2PChannel) Fd() int {
```
Comment (zolia) [inline core/connection/manager_test.go:571]:
```suggestion
func (m *mockP2PChannel) FD() int {
```

File: p2p/channel.go [hunk 3]
```diff
@@ -31,16 +31,28 @@ import (
...
+// Channel represents p2p communication channel which can send and receive messages over encrypted and reliable UDP transport.
+type Channel interface {
+	ChannelSender
+	ChannelHandler
 	// ServiceConn returns UDP connection which can be used for services.
 	ServiceConn() *net.UDPConn
 	// Close closes p2p communication channel.
 	Close() error
+	// Fd returns underlying connection file descriptor which can be used to protect it from VPN tunnel.
+	Fd() int
```
Comment (zolia) [inline p2p/channel.go:55]:
```suggestion
	FD() int
```

File: p2p/channel.go [hunk 4]
```diff
@@ -114,7 +138,8 @@ func (c *channel) readLoop() {
 		default:
 			var msg transportMsg
 			if err := msg.readFrom(c.conn); err != nil {
-				continue
+				log.Debug().Err(err).Msgf("Closing p2p read loop")
```
Comment (zolia) [inline p2p/channel.go:141]:
```suggestion
				log.Debug().Err(err).Msg("Closing p2p read loop")
```

File: core/connection/manager_test.go (hunk 2 above)
Comment (anjmao) [reply core/connection/manager_test.go:571]: I wanted to keep method the same as in go std. `func (f *File) Fd() uintptr`

**Pull request merged**